packages/data/addon/serializers/metadata/metric-function.ts

```diff
@@ -0,0 +1,84 @@
+/**
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ *
+ * Description: A serializer for the metric function endpoint
+ */
+
+import EmberObject from '@ember/object';
+import { RawMetricFunctionArguments, RawMetricFunctionArgument, RawMetricFunction } from './bard';
+import { MetricFunctionMetadataPayload } from 'navi-data/models/metadata/metric-function';
+import {
+  FunctionArgumentMetadataPayload,
+  INTRINSIC_VALUE_EXPRESSION
+} from 'navi-data/models/metadata/function-argument';
+
+type RawMetricFunctionEnvelope = {
+  'metric-functions': {
+    rows: RawMetricFunction[];
+  };
+};
+
+/**
+ * @param parameters - raw metric function parameters
+ * @param source - data source name
+ */
+export function constructFunctionArguments(
+  parameters: RawMetricFunctionArguments,
+  source: string
+): FunctionArgumentMetadataPayload[] {
+  return Object.keys(parameters).map(param => {
+    const paramObj: RawMetricFunctionArgument = parameters[param];
+    const { type, defaultValue, values, dimensionName, description } = paramObj;
```

Comment:
```suggestion
    const { type, defaultValue, values, dimensionName, description }: RawMetricFunctionArgument = parameters[param];
```
Can we do this with typescript?

Reply:
TS knows the shape from line 31.

Reply:
Yeah but we're not using `paramObj` anywhere else so it seems like an unnecessary assignment.